In [13]:
import pandas as pd
from datetime import datetime

In [14]:
df_PM_activities = pd.read_csv('PM_activities.csv')

In [15]:
df = df_PM_activities.drop(columns=['Unnamed: 0', 'number'])
df['publish_date'] = pd.to_datetime(df['publish_date'])
df['date_est'] = df['publish_date'] - pd.Timedelta(days=1)
df.to_csv("df.csv",encoding='UTF-8')
df.head(3)

original          PM    date  \
0  \n12月26日（首相官邸）\n2012/12/27 日本経済新聞　朝刊 4ページ 197文...  Shinzo_Abe  12月26日   
1  \n12月27日（首相官邸）\n2012/12/28 日本経済新聞　朝刊 4ページ 261文...  Shinzo_Abe  12月27日   
2  \n12月28日（首相官邸）\n2012/12/29 日本経済新聞　朝刊 4ページ 476文...  Shinzo_Abe  12月28日   

  publish_date                                         activities   date_est  
0   2012-12-27  〈26日〉▽18時22分　皇居で首相親任式、閣僚認証式。▽21時39分　官邸で記者会見。▽2... 2012-12-26  
1   2012-12-28  ▽10時21分　官邸で菅官房長官、加藤官房副長官。▽11時29分　古屋拉致担当相、三谷拉致問... 2012-12-27  
2   2012-12-29  ▽8時40分　富ケ谷の私邸発。▽9時1分　党本部で高村副総裁、石破幹事長。▽9時34分　官邸... 2012-12-28

In [4]:
from datetime import datetime

abe_term = datetime(2020,9,16) - datetime(2012,12,26)
suga_term = datetime(2021,10,4) - datetime(2020,9,16)
kishida_term = datetime(2023,6,30) - datetime(2021,10,4)
abe_term_days = abe_term.days +1
suga_term_days= suga_term.days +1
kishida_term_days= kishida_term.days +1
print(abe_term_days)
print(suga_term_days)
print(kishida_term_days)

2822
384
635


# Search word

In [8]:
word = "政府専用機"
word

'政府専用機'

In [6]:
filtered_df = df[df['activities'].str.contains(word)]
filtered_df[['PM', 'date', 'publish_date', 'activities']].to_csv(f"PM_activities{word}.csv",encoding='UTF-8')
filtered_df[['PM', 'date', 'publish_date', 'activities']]

PM     date publish_date  \
7         Shinzo_Abe     1月3日   2013-01-04   
65        Shinzo_Abe     3月3日   2013-03-04   
85        Shinzo_Abe    3月23日   2013-03-24   
128       Shinzo_Abe  5月6日・7日   2013-05-08   
222       Shinzo_Abe    8月10日   2013-08-11   
...              ...      ...          ...   
2975  Yoshihide_Suga    4月30日   2021-05-01   
3210   Fumio_Kishida   12月28日   2021-12-29   
3431   Fumio_Kishida    8月14日   2022-08-15   
3433   Fumio_Kishida    8月16日   2022-08-17   
3655   Fumio_Kishida     4月2日   2023-04-03   

                                             activities  
7     ▽8時2分　千葉県市原市の森永高滝カントリー倶楽部で親族らとゴルフ。▽16時　グランドハイア...  
65    ▽7時52分　山梨県鳴沢村の別荘発。▽8時28分　静岡県小山町の東富士カントリークラブで昭恵...  
85    ▽7時15分　富ケ谷の私邸発。▽8時36分　神奈川県茅ケ崎市のスリーハンドレッドクラブで親族...  
128   〈6日〉▽7時7分　山梨県鳴沢村の別荘発。▽7時11分　富士河口湖町の富士桜カントリー倶楽部...  
222   ▽7時8分　静養先の山梨県鳴沢村の別荘発。▽7時14分　鳴沢ゴルフ倶楽部で友人、秘書官とゴル...  
...                                                 ...  
2975  ▽7時52分　東京・赤坂の衆院議員宿舎発。56分　官邸。敷地内を散歩。▽9時1分　閣議。25...  
3210  ▽9時4分　公邸から官邸。5分　鈴木宗男日本維新の会参院議員。18分　孤独・孤立対策推進会議...  
3431  ▽10時26分　東京・赤坂の「エースゴルフクラブ赤坂」。ゴルフの練習。裕子夫人同行。▽11時...  
3433  ▽7時4分　公邸発。59分　茨城県つくばみらい市のゴルフ場「取手国際ゴルフ倶楽部」。裕子夫人...  
3655  ▽6時12分　公邸発。▽7時3分　茨城県つくばみらい市のゴルフ場「筑波カントリークラブ」。野...  

[112 rows x 4 columns]

In [7]:
abe_count = df[df['PM'] == 'Shinzo_Abe']['activities'].str.contains(word).sum() 
suga_count = df[df['PM'] == 'Yoshihide_Suga']['activities'].str.contains(word).sum() 
kishida_count = df[df['PM'] == 'Fumio_Kishida']['activities'].str.contains(word).sum() 

abe_freq = abe_count/abe_term_days *365
suga_freq = suga_count/suga_term_days *365
kishida_freq = kishida_count/kishida_term_days *365

print(word)
print(f"Shinzo_Abe: {abe_count} times; {round(abe_freq,2)}/year")
print(f"Yoshihide_Suga:{suga_count} times; {round(suga_freq,2)}/year")
print(f"Fumio_Kishida:{kishida_count} times; {round(kishida_freq,2)}/year")


ゴルフ
Shinzo_Abe: 107 times; 13.84/year
Yoshihide_Suga:1 times; 0.95/year
Fumio_Kishida:4 times; 2.3/year


散髪
Shinzo_Abe: 93 times; 12.03/year
Yoshihide_Suga:12 times; 11.41/year
Fumio_Kishida:42 times; 24.14/year

散歩
Shinzo_Abe: 16 times; 2.07/year
Yoshihide_Suga:199 times; 189.15/year
Fumio_Kishida:0 times; 0.0/year

マッサージ
Shinzo_Abe: 1 times; 0.13/year
Yoshihide_Suga:2 times; 1.9/year
Fumio_Kishida:10 times; 5.75/year

ゴルフ
Shinzo_Abe: 107 times; 13.84/year
Yoshihide_Suga:1 times; 0.95/year
Fumio_Kishida:4 times; 2.3/year

政府専用機
Shinzo_Abe: 292 times; 37.77/year
Yoshihide_Suga:13 times; 12.36/year
Fumio_Kishida:56 times; 32.19/year

インタビュー
Shinzo_Abe: 751 times; 97.14/year
Yoshihide_Suga:135 times; 128.32/year
Fumio_Kishida:262 times; 150.6/year